src/java.base/share/classes/jdk/internal/classfile/impl/BufWriterImpl.java

```diff
@@ -93,12 +96,28 @@ public ClassFileImpl context() {
         return context;
     }
 
+    void writeMagic(int minorVersion, int majorVersion) {
```

Comment:
Can we use `writeLong(((long) MAGIC_NUMBER) << 32 | minorVersion << 16 | majorVersion);`?